lib/live_select/component.ex

```diff
@@ -168,7 +168,19 @@ defmodule LiveSelect.Component do
       if Map.has_key?(assigns, :value) do
         update(socket, :selection, fn
           selection, %{options: options, value: value, mode: mode, value_mapper: value_mapper} ->
-            update_selection(value, selection, options, mode, value_mapper)
+            set_selection(value, selection, options, mode, value_mapper)
+        end)
+        |> client_select(%{input_event: true})
+      else
+        socket
+      end
+
+    socket =
+      if Map.has_key?(assigns, :update_selection) do
+        update(socket, :selection, fn
+          selection,
+          %{update_selection: update_fn, options: options, mode: mode, value_mapper: value_mapper} ->
+            update_selection(update_fn, selection, options, mode, value_mapper)
```

Comment:
My suggestion is to replace this line with:

```
update_selection(update_fn.(selection), selection, options, mode, value_mapper)
```

Where `update_selection` is the old `update_selection` function, we don't rename it and we don't add the new `update_selection` function.

We won't have dupe detection but I think it's overkill. We don't have it anyway when you pass the selection explicitly with `:value`, why have it in the case of `:update_selection`? The caller can easily check for dupes and remove them if they want to

Reply:
I ran into a problem with this and still need to think of a way to figure it out. The `update_fn.(selection)` may leave mixed entries (current + new) and it fails when trying to ran the existing ones with the `value_mapper` again. This is why the function I wrote splits it and only "normalize" the new entries.

Reply:
Sorry that was not clear, my function fixes the problem by splitting it and selectively running `value_mapper` only on new values. If we are to revert and use the og `update_selection` we must address the `value_mapper` that may encounter already mapped values.

Reply:
Thanks for clarifying. Ok, but I don't think the problem is actually fixed by your function. Take this example:

```elixir
current_selection = [%{value: 1, label: "one"}, %{value: 2, label: "two"}]
value_mapper = fn n -> %{value: n, label: to_string(n)} end

update_fn = fn current_selection -> current_selection ++ [2, 3] end
```

With this arguments, your `update_selection` function 
will fail to spot that "2" is already in the selection, because it's looking for unmapped values (2,3) in the list of mapped values (line 547).

I can't think of an elegant solution for this. Can you?

My impulse would be to say: it's the responsibility of the writer of `update_fn` to leave the selection in a state that can be mapped using the `value_mapper` they provided.
Same for dupe detection.

Reply:
Ok I thought about it and I think another, perhaps better option would to not call `value_mapper` at all when updating the selection programmatically. This makes sense because the caller can map the selection themselves if they want to (i.e. they can provide the selection already in the right format). `value_mapper` was intended anyway  for the case where the user has no way to map the selection (i.e. when it's coming from existing values in the form).

So something like:
```
update_selection(update_fn.(selection), selection, options, mode, & &1)
```